test/lib/rubycritic/revision_comparator_test.rb

```diff
@@ -0,0 +1,63 @@
+require 'test_helper'
+require 'rubycritic/revision_comparator'
+
+describe Rubycritic::RevisionComparator do
+  subject { Rubycritic::RevisionComparator.new([]) }
+
+  describe '#set_statuses' do
+    context 'in a SCS with :revision? == false' do
+      before do
+        Rubycritic::Config.expects(:source_control_system)
+                          .at_least_once
+                          .returns(stub(revision?: false))
+      end
+
+      it 'does not attempt to compare with previous results' do
+        subject.expects(:load_cached_analysed_modules).never
+        result = subject.set_statuses([])
+        result.must_equal([])
+      end
+    end
+
+    context 'in a SCS with :revision? == true' do
+      before do
+        Rubycritic::Config.expects(:source_control_system)
+                          .at_least_once
+                          .returns(stub(revision?: true))
+      end
+
+      context 'without previously cached results' do
+        before do
+          subject.expects(:revision_file).returns('foo')
+          File.expects(:file?).with('foo').returns(false)
+        end
+
+        it 'attempts to load previously cached results' do
```

Comment:
Sorry for not being clear here: My problem with those tests are that this context says:

>> 'without previously cached results'

and the example says:

>> 'attempts to load previously cached results'

From a logical point of view this doesn't seem to make sense. If there are no cached results it shouldn't try to load them.
Apart from that below is a context that says:

>> 'with previously cached results'

but has the **exact** same example:

>> 'attempts to load previously cached results'

So I'm not sure what's going on here, but if I just read the specs as an innocent reader this is very confusing to me.